torchvision/datasets/folder.py

```diff
@@ -38,6 +38,24 @@ def make_dataset(
     extensions: Optional[Tuple[str, ...]] = None,
     is_valid_file: Optional[Callable[[str], bool]] = None,
 ) -> List[Tuple[str, int]]:
+    """Generates a list of samples of a form (path_to_sample, class).
+
+    Args:
+        directory (str): root dataset directory
+        class_to_idx (Dict[str, int]): dictionary mapping class name to class index
+        extensions (Optional[Tuple[str, ...]], optional): A list of allowed extensions.
```

Comment:
Optional is repeated twice now (here and below). I'm not sure if we started to add arg type as typing annotations... @pmeier what do you think ?

Reply:
I think we should stick to one form. Either

```rst
extensions (optional): A list of allowed extensions.
```

or

```rst
extensions (Optional[Tuple[str, ...]]): Allowed extensions.
```

I prefer the second version especially because the type hints can be auto generated from the annotations. That being said we need a `sphinx` extension for that and thus we should stick to the first version for now. 

---

I've just realized that the type annotation and the docstring contradict each other. Should this be a `list` or a `tuple`?

Reply:
Fixed.

the latter type was auto-generated, sorry about that. Pushed the changes that hopefully resolve this :)